signing.go

```diff
@@ -48,32 +55,35 @@ func SignJSON(signingName string, keyID KeyID, privateKey ed25519.PrivateKey, me
 		signatures = map[string]map[KeyID]Base64String{}
 	}
 
+	// Encode the JSON object without the "signatures" key or
+	// the "unsigned" key in the canonical format.
 	unsorted, err := json.Marshal(object)
 	if err != nil {
 		return nil, err
 	}
-
 	canonical, err := CanonicalJSON(unsorted)
 	if err != nil {
 		return nil, err
 	}
 
+	// Sign the canonical JSON with the ed25519 key.
 	signature := Base64String(ed25519.Sign(privateKey, canonical))
```

Comment:
I'm usually against adding comments with so much redundant information, but I think it's useful here to break up the function into sections.